dask/highlevelgraph.py

```diff
@@ -594,8 +594,8 @@ def __getitem__(self, key):
 
         raise KeyError(key)
 
-    def __len__(self):
-        return len(self.to_dict())
+    def __len__(self) -> int:
+        return sum(len(layer) for layer in self.layers.values())
```

Comment:
pls add comment explaining how this could double-count keys but we decided not to care as it should always be a broken use case

Reply:
I was actually wondering about this when I looked at this PR last night, but decided not to say anything.  I'm glad to hear that this already came up and has been considered.

Reply:
@mrocklin we discussed it in #7271; i'd like to hear your input on it though